projects/igniteui-angular/src/lib/grids/watch-changes.ts

```diff
@@ -0,0 +1,31 @@
+import { SimpleChanges, SimpleChange } from '@angular/core';
+
+/**
+* @hidden
+*/
+export function WatchChanges(): PropertyDecorator {
```

Comment:
Guess who's back, back again...

https://www.youtube.com/watch?v=YVkUvmDQ3HY

Reply:
The idea behind this prop decorator is the following – if the marked property is changed directly via its setter it triggers the ngOnChanges with the related change information. 
This is done so components like the RowIsland that needs to track and delegate input property changes to its related child grid components (both existing and not yet existing) can ‘watch’ if an input property setter was invoked with a new value and if so trigger ngOnChanges so that the change can be handled accordingly.
This does not trigger change detection. Just makes sure that the changes event is triggered regardless of whether a prop change was done as a result of angular’s model binding or if the setter was directly called.